app/database/admin/data_operator/index.ts

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {MM_TABLES} from '@constants/database';
+import {
+    BatchOperations,
+    DataFactory,
+    DBInstance,
+    HandleBaseData,
+    HandleIsolatedEntityData,
+    RecordValue,
+} from '@typings/database/database';
+
+import DatabaseManager from '../database_manager';
+
+import {
+    operateAppRecord,
+    operateCustomEmojiRecord,
+    operateGlobalRecord,
+    operateRoleRecord,
+    operateServersRecord,
+    operateSystemRecord,
+    operateTermsOfServiceRecord,
+} from './operators';
+
+export enum OperationType {
+    CREATE = 'CREATE',
+    UPDATE = 'UPDATE',
+    DELETE = 'DELETE'
+}
+
+class DataOperator {
+    private defaultDatabase: DBInstance;
+    private serverDatabase: DBInstance;
+
+    /**
+     * handleIsolatedEntityData: Operator that handles Create/Update operations on the isolated entities as
+     * described by the IsolatedTables type
+     * @param {OperationType} optType
+     * @param {APP | GLOBAL | SERVERS | CUSTOM_EMOJI | ROLE | SYSTEM | TERMS_OF_SERVICE} tableName
+     * @param {RawApp | RawGlobal | RawServers} values
+     * @returns {Promise<void>}
+     */
+    handleIsolatedEntityData = async ({optType, tableName, values}: HandleIsolatedEntityData): Promise<void> => {
+        const {APP, GLOBAL, SERVERS} = MM_TABLES.DEFAULT;
+        const {CUSTOM_EMOJI, ROLE, SYSTEM} = MM_TABLES.SERVER;
+
+        let recordOperator: (recordOperator: DataFactory) => void;
+
+        switch (tableName) {
+        case APP : {
+            recordOperator = operateAppRecord;
+            break;
+        }
+        case GLOBAL : {
+            recordOperator = operateGlobalRecord;
+            break;
+        }
+        case SERVERS : {
+            recordOperator = operateServersRecord;
+            break;
+        }
+        case CUSTOM_EMOJI : {
+            recordOperator = operateCustomEmojiRecord;
+            break;
+        }
+        case ROLE : {
+            recordOperator = operateRoleRecord;
+            break;
+        }
+        case SYSTEM : {
+            recordOperator = operateSystemRecord;
+            break;
+        }
+        default: {
+            // TERMS_OF_SERVICE
```

Comment:
Question: is the intent to have terms of service operator by default even if the table name passed is not terms of service or misspelled? Or should this be its own case and default to throw an exception for an incorrect table?

Reply:
Might be good to add tests for all cases to make the right recordOperator is being assigned

Reply:
Thanks a lot @josephbaylon .  I will fix both this comment and the one below as they are related.

Reply:
Corrected.

Reply:
Thanks! Please add a test for handling default case